index.js

```diff
@@ -1,16 +1,47 @@
 'use strict';
 module.exports = function (str) {
-	return str.replace(/(?:\\*)?"([^"\\]*\\.)*[^"]*"/g, function (match) {
-		return match
-			// unescape double-quotes
-			.replace(/([^\\]|^)\\"/g, '$1"')
-			// escape escapes
-			.replace(/(^|[^\\])(\\+)'/g, '$1$2\\\'')
-			// escape single-quotes - round 1
-			.replace(/([^\\])'/g, '$1\\\'')
-			// escape single-quotes - round 2 (for consecutive single-quotes)
-			.replace(/([^\\])'/g, '$1\\\'')
-			// convert
-			.replace(/^"|"$/g, '\'');
-	});
+	var result = '';
+	var x;
+	var len = str.length;
```

Comment:
@sindresorhus Thank you for the feedback! Just pushed the new version with the requested changes.